src/Query/SemanticSearch/ChainDocumentsTransformer.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace LLPhant\Query\SemanticSearch;
+
+class ChainDocumentsTransformer implements RetrievedDocumentsTransformer
+{
+    /** @var RetrievedDocumentsTransformer[] */
+    private array $transformers = [];
+
+    public function addTransformer(RetrievedDocumentsTransformer $retrievedDocumentsTransformer = new IdentityDocumentsTransformer()): void
```

Comment:
What do you think about adding this method to `PipeDocumentsTransformer`?

Reply:
- merged that method into PipeDocumentsTransformer
- removed ChainDocumentsTransformer as it seems redundant
- renamed ChainTransformer to PipeTransformer to match naming scheme
- added a PipeTransformer constructor similar to PipeDocumentsTransformer